src/SPC/command/BuildPHPCommand.php

```diff
@@ -158,14 +159,24 @@ public function handle(): int
             if ($this->input->getOption('with-upx-pack') && in_array(PHP_OS_FAMILY, ['Linux', 'Windows'])) {
                 $indent_texts['UPX Pack'] = 'enabled';
             }
+
             try {
-                $ver = $builder->getPHPVersion();
-                $indent_texts['PHP Version'] = $ver;
-            } catch (\Throwable) {
-                if (($ver = $builder->getPHPVersionFromArchive()) !== false) {
-                    $indent_texts['PHP Version'] = $ver;
+                $cleanPhpSrc = $builder->getPHPVersion() !== $builder->getPHPVersionFromArchive();
+            } catch (RuntimeException|WrongUsageException) {
+                $cleanPhpSrc = true;
+            }
+            if ($cleanPhpSrc) {
+                logger()->info('Cleaning previous php build due to mismatching versions...');
+                FileSystem::removeDir(SOURCE_PATH . '/php-src');
+                FileSystem::removeDir(BUILD_MODULES_PATH);
```

Comment:
@crazywhalecc do we still need this after your lock changes?

Reply:
No need.